File: src/main/java/com/core/base/corebase/domain/organization/Team.java
package com.core.base.corebase.domain.organization;

import com.core.base.corebase.domain.common.BaseDateTimeEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "team")
public class Team extends BaseDateTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(name = "order_num", nullable = false)
    private Integer orderNum;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_id", insertable = false, updatable = false)
    private Team parents;

    @Column(name = "parent_id")
    private Long parentId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "organization_id", nullable = false)
    private Organization organization;

    protected Team() {
        // For JPA
    }

    public Team(String name, Integer orderNum, Long parentId, Organization organization) {
        this.name = name;
        this.orderNum = orderNum;
        this.parentId = parentId;
        this.organization = organization;
    }

    public void update(String name, Integer orderNum, Long parentId) {
        this.name = name;
        this.orderNum = orderNum;
        this.parentId = parentId;
    }

    public void update(Integer orderNum, Long parentId) {
        this.orderNum = orderNum;
        this.parentId = parentId;
    }
}

File: src/main/kotlin/com/core/base/corebase/domain/organization/Team.kt
package com.core.base.corebase.domain.organization

import jakarta.persistence.*
import net.huray.backend.minuting.entity.common.BaseDateTimeEntity

@Entity(name ="team")
class Team(
    name: String,
    order: Int,
    parentId: Long?,
    organization: Organization
): BaseDateTimeEntity() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0L

    var name = name; protected set
    var order = order; protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_id", insertable = false, updatable = false)
    var parents: Team? = null
        protected set

    @Column(name = "parent_id", nullable = true)
    var parentId: Long? = parentId

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "organization_id", nullable = false)
    var organization: Organization = organization; protected set


    fun update(name: String, order: Int, parentId: Long?) {
        this.name = name
        this.order = order
        this.parentId = parentId
    }
}